src/binary/object.hpp

```diff
@@ -1,14 +1,16 @@
 #ifndef OBJECT_HPP
 #define OBJECT_HPP
 
-#include "utils/common.hpp"
-#include "utils/utils.hpp"
-#include "binary/module_base.hpp"
-
-#include <string>
 #include <vector>
+#include <cstdint>
 
 namespace cpptrace {
+
+struct object_frame;
+struct safe_object_frame;
+
+using frame_ptr = std::uintptr_t;
```

Comment:
Can we `#include <cpptrace/cpptrace.hpp>` here instead. My main concern is this falling out of line with the definition there.

Reply:
I measured this change making a noticeable difference via `ClangBuildAnalyzer`. If there is a mismatch between these forward declarations and `cpptrace.hpp`, compilation will fail with an hard error (i.e. you won't silently introduce any UB).

Given that, would you reconsider?

Reply:
I think that’s reasonable, I appreciate you measuring. I had also been thinking of splitting ip the cpptrace.hpp file so one option in the future might be to have a more minimal header providing the type.